consensus/src/runway/collection.rs

```diff
@@ -168,15 +168,26 @@ impl<'a, MK: KeyBox> Collection<'a, MK> {
         if response.salt != self.salt {
             return Err(Error::SaltMismatch(self.salt, response.salt));
         }
-        if let Some(unchecked_unit) = response.unit {
-            let checked_signed_unit = self.validator.validate_unit(unchecked_unit)?;
-            let checked_unit = checked_signed_unit.as_signable();
-            if checked_unit.creator() != self.keychain.index() {
-                return Err(Error::ForeignUnit(checked_unit.creator()));
+        let round: Round = match response.unit {
+            Some(unchecked_unit) => {
+                let checked_signed_unit = self.validator.validate_unit(unchecked_unit)?;
+                let checked_unit = checked_signed_unit.as_signable();
+                if checked_unit.creator() != self.keychain.index() {
+                    return Err(Error::ForeignUnit(checked_unit.creator()));
+                }
+                checked_unit.round() + 1
             }
-            self.starting_round = max(self.starting_round, checked_unit.round() + 1);
+            None => 0,
+        };
+        let current_round = *self
+            .collected_starting_rounds
+            .get(response.responder)
+            .unwrap_or(&0);
+        if current_round != round {
+            info!(target: "AlephBFT-runway", "Warning: node {} responded with starting unit {}, but now says {}", response.responder.0, current_round, round);
```

Comment:
I really don't think this is necessary, if you want something here I would put `warn` here, but only if `rounds < current_round`, because that is actually unexpected.

Reply:
@DamianStraszak - what do you think?

Reply:
You can either remove it, or make it a `warn` (and remove the inner `Warning`). I would say it's rather unexpected, to the extent that I prefer to keep it as an interesting bit of info.

Reply:
changed `info` to `warn`, keeping the condition unchanged

Reply:
That's not in accordance with our logging standards – this `warn` can now appear during normal operation with no adversaries. From the AlephBFT point of view because we cannot be sure that the other node did not belatedly receive a higher unit and, more practically, from the aleph-node point of view because of data availability. I would say if `round > current_round` then this cannot be a warning, and preferably would be a `debug` at most, if not completely ignored (my preferred approach).

Reply:
I'm ok with removing or turning into `debug`.

Reply:
leaving this message as `debug`